controllers/util/provider.go

```diff
@@ -147,6 +159,19 @@ func GetProviderCredentials(ctx context.Context, k8sClient client.Client, namesp
 				EnvARMSubscriptionID: cred.ARMSubscriptionID,
 				EnvARMTenantID:       cred.ARMTenantID,
 			}, nil
+		case string(VSphere):
+			var cred VSphereCredentials
+			if err := yaml.Unmarshal(secret.Data[secretRef.Key], &cred); err != nil {
+				errMsg := "failed to convert the credentials of Secret from Provider"
+				klog.ErrorS(err, errMsg, "Name", secretRef.Name, "Namespace", secretRef.Namespace)
+				return nil, errors.Wrap(err, errMsg)
+			}
+			return map[string]string{
+				EnvVSphereUser:               cred.VSphereUser,
+				EnvVSpherePassword:           cred.VSpherePassword,
+				EnvVSphereServer:             cred.VSphereServer,
+				EnvVSphereAllowUnverifiedSSL: cred.VSphereAllowUnverifiedSSL,
```

Comment:
As `VSphereAllowUnverifiedSSL` is optional, can you test it without the corresponding environment variable?

Reply:
my vsphere server is self-signed cert.
When I don't set VSPHERE_ALLOW_UNVERIFIED_SSL or set to false, it will have an error of x509. ( certificate signed by unknown authority )
when i set it is true, everything is ok.

Reply:
Okay, let's keep it as `required` at present.